torchserve/datadog_checks/torchserve/management_api_check.py

```diff
@@ -0,0 +1,184 @@
+        self.model_discovery = None
+        self.submit_events = None
+        self.tags = None
+        self.__previous_models = None
+        self.check_initializations.append(self.parse_config)
+
+    def parse_config(self):
+        self.model_discovery = ModelDiscovery(
+            self,
+            limit=self.instance.get("limit", 100),
+            include=self.instance.get("include", [".*"]),
+            exclude=self.instance.get("exclude", []),
+            interval=self.instance.get("interval"),
+        )
+        self.submit_events = self.instance.get("submit_events", True)
+        self.tags = self.instance.get("tags", [])
+        self.tags.append(f"management_api_url:{self.instance['management_api_url']}")
+
+    def check(self, _):
+        for _, _, model, _ in self.model_discovery.get_items():
+            self.process_one_model(model)
+
+        self.service_check(self.SERVICE_CHECK_NAME, self.model_discovery.api_status, tags=self.tags)
+
+        if self.model_discovery.api_status == AgentCheck.OK and self.submit_events and self.model_discovery.refreshed:
+            if self.previous_models is not None:
+                self.compare_models_and_emit_events(self.previous_models, self.model_discovery.all_models)
+
+            self.previous_models = self.model_discovery.all_models
+
+    def process_one_model(self, model):
+        self.log.debug("Processing model [%s]", model['modelName'])
+        model_name = model['modelName']
+        endpoint = f"models/{model_name}/all"
+        url = urljoin(self.instance["management_api_url"], endpoint)
+
+        try:
+            response = self.http.get(url)
+            response.raise_for_status()
+            content = response.json()
+        except Exception as e:
+            self.log.error("Caught exception %s querying model %s.", e, model_name)
+            return
+
+        self.gauge("model.versions", len(content), tags=self.tags + [f"model_name:{model_name}"])
+
+        for model_details in content:
+            self.process_one_version(model, model_details)
+
+    def process_one_version(self, model, model_details):
+        self.log.debug(
+            "Processing version [%s] for model [%s]", model_details['modelVersion'], model_details['modelName']
+        )
+
+        tags = copy.deepcopy(self.tags)
+        tags += [f"model_name:{model_details['modelName']}", f"model_version:{model_details['modelVersion']}"]
+
+        self.gauge("model.workers.current", len(model_details.get("workers", [])), tags=tags)
+        self.gauge("model.workers.min", model_details.get("minWorkers"), tags=tags)
+        self.gauge("model.workers.max", model_details.get("maxWorkers"), tags=tags)
+        self.gauge("model.batch_size", model_details.get("batchSize"), tags=tags)
+        self.gauge("model.max_batch_delay", model_details.get("maxBatchDelay"), tags=tags)
+        self.gauge("model.is_loaded_at_startup", 1 if model_details.get("loadedAtStartup", False) else 0, tags=tags)
```

Comment:
No need to address this. For brevity this can also be represented as 

`int(model_details.get("loadedAtStartup", False))` or `model_details.get("loadedAtStartup", False) * 1`

:)

Reply:
That's nice! but I'm not sure that's easier to read